metrics/telemetry.go

```diff
@@ -11,13 +11,50 @@ import (
 	"net/http"
 
 	"github.com/omec-project/ausf/logger"
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
+// AusfStats captures AUSF stats
+type AusfStats struct {
+	ueAuths *prometheus.CounterVec
+}
+
+var ausfStats *AusfStats
+
+func initAusfStats() *AusfStats {
+	return &AusfStats{
+		ueAuths: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Name: "ausf_ue_authentications_total",
+			Help: "Counter of total UE Authentications",
+		}, []string{"ausf_id", "serving_network_name", "auth_type", "result"}),
+	}
+}
+
+func (ps *AusfStats) register() error {
+	if err := prometheus.Register(ps.ueAuths); err != nil {
+		return err
+	}
+	return nil
+}
+
+func init() {
+	ausfStats = initAusfStats()
+
+	if err := ausfStats.register(); err != nil {
+		logger.InitLog.Panicln("AUSF Stats register failed")
+	}
+}
+
 // InitMetrics initialises AUSF metrics
 func InitMetrics() {
 	http.Handle("/metrics", promhttp.Handler())
 	if err := http.ListenAndServe(":8080", nil); err != nil {
```

Comment:
Hi @dariofaccin, is it possible that you help us parametrizing the port number for the metrics across all NFs? There is interests from users to start getting metrics from different NFs. So, it would be good to have this parametrized. Thanks!